pkg/cmd/openshift-service-serving-cert-signer/config.go

```diff
@@ -0,0 +1,214 @@
+	if syntaxErr, ok := err.(*json.SyntaxError); err != nil && ok {
+		offset := syntaxErr.Offset
+		begin := offset - 20
+		if begin < 0 {
+			begin = 0
+		}
+		end := offset + 20
+		if end > int64(len(data)) {
+			end = int64(len(data))
+		}
+		return fmt.Errorf("%s%v (found near '%s')", prefix, err, string(data[begin:end]))
+	}
+	if err != nil {
+		return fmt.Errorf("%s%v", prefix, err)
+	}
+	return err
+}
+
+func resolveConfigPaths(config *v1alpha1.ServiceServingCertSignerConfig, base string) error {
+	return resolvePaths(getFileReferences(config), base)
+}
+
+func getFileReferences(config *v1alpha1.ServiceServingCertSignerConfig) []*string {
+	refs := []*string{}
+
+	refs = append(refs, &config.ServingInfo.CertInfo.CertFile)
+	refs = append(refs, &config.ServingInfo.CertInfo.KeyFile)
+	refs = append(refs, &config.ServingInfo.ClientCA)
+	for i := range config.ServingInfo.NamedCertificates {
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].CertFile)
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].KeyFile)
+	}
+	refs = append(refs, &config.Signer.CertFile)
+	refs = append(refs, &config.Signer.KeyFile)
+
+	return refs
+}
+
+// resolvePaths updates the given refs to be absolute paths, relative to the given base directory
+func resolvePaths(refs []*string, base string) error {
+	for _, ref := range refs {
+		// Don't resolve empty paths
+		if len(*ref) > 0 {
+			// Don't resolve absolute paths
+			if !filepath.IsAbs(*ref) {
+				*ref = filepath.Join(base, *ref)
+			}
+		}
+	}
+	return nil
+}
+
+type ClientConnectionOverrides struct {
+	// AcceptContentTypes defines the Accept header sent by clients when connecting to a server, overriding the
+	// default value of 'application/json'. This field will control all connections to the server used by a particular
+	// client.
+	AcceptContentTypes string
+	// ContentType is the content type used when sending data to the server from this client.
+	ContentType string
+
+	// QPS controls the number of queries per second allowed for this connection.
+	QPS float32
+	// Burst allows extra queries to accumulate when a client is exceeding its rate.
+	Burst int32
+}
+
+var defaultConnectionOverrides = &ClientConnectionOverrides{
+	AcceptContentTypes: "application/vnd.kubernetes.protobuf,application/json",
+	ContentType:        "application/vnd.kubernetes.protobuf",
+	QPS:                50,
+	Burst:              100,
+}
+
+// getKubeConfigOrInClusterConfig loads in-cluster config if kubeConfigFile is empty or the file if not,
+// then applies overrides.
+func getKubeConfigOrInClusterConfig(kubeConfigFile string, overrides *ClientConnectionOverrides) (*rest.Config, error) {
+	if len(kubeConfigFile) > 0 {
+		return getClientConfig(kubeConfigFile, overrides)
+	}
+
+	clientConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	applyClientConnectionOverrides(overrides, clientConfig)
+	clientConfig.WrapTransport = defaultClientTransport
```

Comment:
While functionally not different at the moment, can we set the default here before applying overrides? If we do add custom transports (such as we do [here](https://github.com/kubernetes/kubernetes/commit/6463e9efd9ba552e60d2555a3e6526ef90196473#diff-0861e1cd492e078057f9ec7524d9d6ffR156) for the external exec auth plugin), it would be nice to not have to track down where it's getting overridden by the default.